Source/WebCore/Modules/fetch/FetchHeaders.cpp

```diff
@@ -181,13 +207,35 @@ ExceptionOr<String> FetchHeaders::get(const String& name) const
 {
     if (!isValidHTTPToken(name))
         return Exception { TypeError, makeString("Invalid header name: '", name, "'") };
+
+    if (equalIgnoringASCIICase(name, "Set-Cookie"_str)) {
+        if (m_setCookieValues.isEmpty())
+            return String();
```

Comment:
This is just about the size-0 case, right? The size-1 optimization can stay.

Reply:
Oh, I hadn't realized that returning `String()` on the size-0 case isn't an optimization, since `get()` must return `null`. The size-1 case is an optimization, but not apparently one that would achieve anything, since it seems like `StringBuilder` directly uses the passed string in this case.

I'm therefore keeping the size-0 case and removing the size-1 case.